### PR TITLE
Update init test and migrate imports for plugin framework

### PR DESCRIPTION
**Pull request by edwardfeng-db**
## Changes

- Part of https://github.com/databricks/terraform-provider-databricks/pull/3841
- Migrated imports following the documentation [here](https://developer.hashicorp.com/terraform/plugin/testing/migrating)
  - 
image

- Made `run` in `init_test` able to take both plugin framework resources as well as sdkv2 resources

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: internal/acceptance/init_test.go [hunk 1]
```diff
@@ -138,7 +143,38 @@ func run(t *testing.T, steps []step) {
...
+				},
+			}
+
+			muxServer, err := tf6muxserver.NewMuxServer(ctx, providers...)
+
+			if err != nil {
+				return nil, err
+			}
+
+			return muxServer.ProviderServer(), nil
```
Comment (mgyucht) [inline internal/acceptance/init_test.go:174]: There should be one definition of this shared by both the main provider and integration tests. Can we combine these together?

File: internal/acceptance/init_test.go [hunk 2]
```diff
@@ -207,10 +242,16 @@ func run(t *testing.T, steps []step) {
...
+
+					// Convert InstanceState from terraform-plugin-framework to terraform-plugin-sdk v2
+					sdkInstanceState := convertToSDKInstanceState(is.Primary)
+
+					if r, ok := sdkPluginProvider.ResourcesMap[p[0]]; ok {
+						// Only checking for sdkv2 resources.
+						dia := r.ReadContext(ctx, r.Data(sdkInstanceState), client)
+						if dia != nil {
+							return fmt.Errorf("%v", dia)
+						}
```
Comment (mgyucht) [inline internal/acceptance/init_test.go:254]: I'm wondering if we really need this at all, even for SDK-based resources. If the test framework refreshes resources to read them again, then I think it is actually OK to remove this entirely. What do you think?

Comment (edwardfeng-db) [reply internal/acceptance/init_test.go:254]: I think maybe for now let's keep it just to minimize the amount of changes we introduce?

File: internal/acceptance/init_test.go (hunk 1 above)
Comment (edwardfeng-db) [reply internal/acceptance/init_test.go:174]: Yeah good point, I can update it

Comment (edwardfeng-db) [reply internal/acceptance/init_test.go]: Updated, PTAL

**Review by mgyucht — APPROVED** [review]
LGTM!

**Pull request merged**